apollo-federation/src/schema/position.rs

```diff
@@ -1808,6 +1833,23 @@ impl ObjectTypeDefinitionPosition {
         self.field(name!("__type"))
     }
 
+    // TODO: Once the new lifetime capturing rules for return position impl trait (RPIT) land in
+    // Rust edition 2024, we will no longer need the "captures" trick here, as noted in
+    // https://rust-lang.github.io/rfcs/3498-lifetime-capture-rules-2024.html
+    pub(crate) fn fields<'a>(
+        &'a self,
+        schema: &'a Schema,
+    ) -> Result<
+        impl Iterator<Item = ObjectFieldDefinitionPosition> + Captures<&'a ()>,
```

Comment:
Does this not work here?
```suggestion
        impl Iterator<Item = ObjectFieldDefinitionPosition> + 'a,
```

Reply:
@goto-bus-stop 
That was the thing I jumped to at first, but [Rust RFC 3498](https://rust-lang.github.io/rfcs/3498-lifetime-capture-rules-2024.html#the-outlives-trick) noted that this is the opposite lifetime constraint we want.